eng/targets/Settings.props

```diff
@@ -51,12 +51,21 @@
     <PublishWindowsPdb>false</PublishWindowsPdb>
     <EnforceExtendedAnalyzerRules>false</EnforceExtendedAnalyzerRules>
     <EnforceExtendedAnalyzerRules Condition="'$(IsAnalyzer)' == 'true'">true</EnforceExtendedAnalyzerRules>
-  </PropertyGroup>
 
-  <PropertyGroup>
-    <SourceBuildTargetFrameworks>net6.0</SourceBuildTargetFrameworks>
-    <SourceBuildTargetFrameworks Condition="'$(ContinuousIntegrationBuild)' == 'true'">net8.0;$(SourceBuildTargetFrameworks)</SourceBuildTargetFrameworks>
-    <SourceBuildTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">net8.0</SourceBuildTargetFrameworks>
+    <!-- 
+      There are effectively three modes that are needed for our source build TFMs
+        - CI normal leg: this needs to build the union of all TFM we support. That ensures we handle all 
+          diagnostics and analyzers for all the TFM that we support.
+        - CI source build leg: this needs to build the current and previous source build TFM. Both are 
+          necessary as the output of this leg is used in other CI source build legs. Those could be 
+          targeting NetCurrent or NetPrevious hence we must produce both.
+        - Source build the product: this is the all up build of the product which needs only NetCurrent
+    -->
+    <SourceBuildTargetFrameworks></SourceBuildTargetFrameworks>
```

Comment:
Nit: seems like this empty property is not needed? (Won't it prevent someone doing `msbuild /p:SourceBuildTargetFrameworks=customvalue` if they want to experiment locally?)

Reply:
> Won't it prevent someone doing msbuild /p:SourceBuildTargetFrameworks=customvalue

It won't prevent that because `/p:` wins all the time as it's a [global property](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-properties?view=vs-2022#global-properties). Essentially it overrides whatever is in the project file and becomes a readonly property of build. It's very counterintuitive. The only way to get around this is to use `TreatAsLocalProperty`. Generally though that feature is not used very often, at least in our builds